### PR TITLE
[SPARK-9424] [SQL] Parquet programming guide updates for 1.5

### PR DESCRIPTION
**Pull request by liancheng**

### CODE REVIEW

**Comment by liancheng** [comment]
@rxin I've updated support for writing decimal values with precision larger than 18 in the release note docs.

File: docs/sql-programming-guide.md [hunk 1]
```diff
@@ -1201,7 +1208,7 @@ saveDF(df1, "data/test_table/key=1", "parquet", "overwrite")
 saveDF(df2, "data/test_table/key=2", "parquet", "overwrite")
 
 # Read the partitioned table
-df3 <- loadDF(sqlContext, "data/test_table", "parquet")
+df3 <- loadDF(sqlContext, "data/test_table", "parquet", mergeSchema="true")
```
Comment (liancheng) [inline docs/sql-programming-guide.md]: @rxin Please help verify whether this change for R is correct, thanks!

File: docs/sql-programming-guide.md [hunk 2]
```diff
@@ -1310,8 +1317,7 @@ Configuration of Parquet can be done using the `setConf` method on `SQLContext`
   <td><code>spark.sql.parquet.int96AsTimestamp</code></td>
   <td>true</td>
   <td>
-    Some Parquet-producing systems, in particular Impala, store Timestamp into INT96. Spark would also
-    store Timestamp as INT96 because we need to avoid precision lost of the nanoseconds field. This
+    Some Parquet-producing systems, in particular Impala and Hive, store Timestamp into INT96.  This
```
Comment (liancheng) [inline docs/sql-programming-guide.md]: Our timestamp type only has microsecond precision now, so updated this note.

**Comment by liancheng** [comment]
cc @marmbrus

File: docs/sql-programming-guide.md (hunk 1 above)
Comment (rxin) [inline docs/sql-programming-guide.md]: cc @shivaram

Comment (shivaram) [inline docs/sql-programming-guide.md]: Yeah thats the right R syntax

**Comment by marmbrus** [comment]
Thanks, merging to master and 1.5

File: docs/sql-programming-guide.md [hunk 3]
```diff
@@ -1371,6 +1380,26 @@ Configuration of Parquet can be done using the `setConf` method on `SQLContext`
...
+  <td><code>false</code></td>
+  <td>
+    <p>
+      When true, the Parquet data source merges schemas collected from all data files, otherwise the
+      schema is picked from the summary file or a random data file if no summary file is available.
+    </p>
+  </td>
+</tr>
+<tr>
+  <td><code>spark.sql.parquet.mergeSchema</code></td>
```
Comment (srowen) [inline docs/sql-programming-guide.md]: CC @liancheng 
@witgo I think you could have just opened a PR to remove this duplicated content as an addendum to SPARK-9424. You opened SPARK-10350 / https://github.com/apache/spark/pull/8520 which is OK, but you didn't tie the two together anywhere. The title and description still don't make this clear.

Comment (liancheng) [inline docs/sql-programming-guide.md]: Agree with @srowen. Michael has already fixed this issue while merging #8520 though.

Comment (witgo) [inline docs/sql-programming-guide.md]: Sorry, next time I will correct.

**Pull request closed without merging**